ext/json/json.c

```diff
@@ -92,11 +92,37 @@ static const zend_function_entry json_serializable_interface[] = {
 };
 /* }}} */
 
+/* {{{ PHP_INI_MH
+ */
+static PHP_INI_MH(OnSetJsonPrecision)
+{
+	zend_long i;
+
+	ZEND_ATOL(i, ZSTR_VAL(new_value));
+	if (i >= -1) {
+		JSON_G(precision) = i;
+		return SUCCESS;
+	} else {
+		return FAILURE;
+	}
+}
+/* }}} */
+
+
+/* {{{ PHP_INI
+ */
+PHP_INI_BEGIN()
+STD_PHP_INI_ENTRY("json.precision", "-1",  PHP_INI_ALL, OnSetJsonPrecision, precision, zend_json_globals, json_globals)
+PHP_INI_END()
```

Comment:
Sure it's more flexible, but imho we should have very strong reasons for introducing new ini settings. More than just "it might be useful to someone ... maybe". Do you have some particular use-case in mind where it would be important to print *inaccurate* JSON output?

If there are indeed cases where controlling the precision of JSON floats is of paramount importance, it might be preferable to add this as an option to json_encode rather than having a global setting for it.